subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/GradleProjectBuilder.java

```diff
@@ -65,7 +66,14 @@ private DefaultGradleProject buildHierarchy(Project project) {
                 .setChildren(children);
 
         gradleProject.getBuildScript().setSourceFile(project.getBuildFile());
-        gradleProject.setTasks(tasks(gradleProject, (TaskContainerInternal) project.getTasks()));
+
+        // Internal system property to investigate model loading performance in IDEA/Android Studio
+        String projectOptions = System.getProperty("org.gradle.unsafe.GradleProjectBuilderOptions", "unmodified");
+        List<LaunchableGradleTask> tasks = tasks(gradleProject, (TaskContainerInternal) project.getTasks(), projectOptions);
+
+        if (!"skip_task_serialization".equals(projectOptions)) {
```

Comment:
Based on benchmark with different sizes of projects, skip_task_serialization doesn't make noticeable differences regarding build time.
I think a flag that skips both of task serialization and realization would be useful, because the task list here is not used by IDEA at all.

Reply:
I've added a new possible value for that.

Reply:
Thank you!